src/pipeline/src/etl.rs

```diff
@@ -32,6 +36,60 @@ pub enum Content {
     Yaml(String),
 }
 
+fn set_processor_keys_index(
+    processors: &mut processor::Processors,
+    final_intermediate_keys: &Vec<String>,
+) -> Result<(), String> {
+    for processor in processors.iter_mut() {
+        for field in processor.fields_mut().iter_mut() {
+            let index = final_intermediate_keys
+                .iter()
+                .position(|r| *r == field.input_field.name)
+                .ok_or(format!(
+                    "input field {} is not found in intermediate keys: {final_intermediate_keys:?} when set processor keys index",
+                    field.input_field.name
+                ))?;
+            field.set_input_index(index);
+            for (k, v) in field.output_fields.iter_mut() {
+                let index = final_intermediate_keys
+                    .iter()
+                    .position(|r| *r == *k)
+                    .ok_or(format!(
+                    "output field {k} is not found in intermediate keys: {final_intermediate_keys:?} when set processor keys index"
+                ))?;
+                *v = index;
+            }
+        }
+    }
+    Ok(())
+}
```

Comment:
Do you think it can be easier to understand if we maintain an `input_field_key_index_mapping` and `output_field_key_index_mapping` in `Processor`?

Reply:
Yes, my plan is to do it in the next step.